pkg/scheduler/flavorassigner/flavorassigner_test.go

```diff
@@ -2172,30 +1797,25 @@ func TestAssignFlavors(t *testing.T) {
 					Request(corev1.ResourceCPU, "9").
 					Obj(),
 			},
-			clusterQueue: cache.ClusterQueue{
-				Cohort: &cache.Cohort{
-					Usage: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "one", Resource: corev1.ResourceCPU}: 2000,
-					}.Unflatten(),
-					RequestableResources: resources.FlavorResourceQuantitiesFlat{
-						{Flavor: "one", Resource: corev1.ResourceCPU}:  10000,
-						{Flavor: "one", Resource: corev1.ResourcePods}: 10,
-					}.Unflatten(),
-				},
-				FlavorFungibility: defaultFlavorFungibility,
-				ResourceGroups: []cache.ResourceGroup{{
-					CoveredResources: sets.New(corev1.ResourceCPU, corev1.ResourcePods),
-					Flavors: []cache.FlavorQuotas{{
-						Name: "one",
-						Resources: map[corev1.ResourceName]*cache.ResourceQuota{
-							corev1.ResourcePods: {Nominal: 10},
-							corev1.ResourceCPU:  {Nominal: 10_000, LendingLimit: ptr.To[int64](0)},
-						},
-					}},
-				}},
-				Usage: resources.FlavorResourceQuantitiesFlat{
+			clusterQueue: utiltesting.MakeClusterQueue("test-clusterqueue").
+				FlavorFungibility(defaultFlavorFungibility).
+				ResourceGroup(
+					utiltesting.MakeFlavorQuotas("one").
+						Resource(corev1.ResourcePods, "10").
+						Resource(corev1.ResourceCPU, "10000m", "10000m", "0m").
+						FlavorQuotas,
+				).Cohort("test-cohort").ClusterQueue,
+			clusterQueueUsage: resources.FlavorResourceQuantitiesFlat{
+				{Flavor: "one", Resource: corev1.ResourceCPU}: 2000,
+			},
+			cohortResources: &cohortResources{
+				usage: resources.FlavorResourceQuantitiesFlat{
 					{Flavor: "one", Resource: corev1.ResourceCPU}: 2000,
-				}.Unflatten(),
+				},
+				requestableResources: resources.FlavorResourceQuantitiesFlat{
+					{Flavor: "one", Resource: corev1.ResourceCPU}:  2000,
```

Comment:
note to reviewers: This test started failing once borrowing limit was added (limitation of the Resource constructor - which requires you to specify borrowing limit to set lending limit)

I think the original test didn't match its intention: since previously there was no borrowing limit, the cq wasn't looking for capacity in its cohort, only looking in cq. I changed this value to 2000m, and specified 10000m borrowing limit.

Reply:
I'm not convinced, if there is no borrowing limit, then the cq is free to borrow within the entire cq.
Would it be feasible to have a new dedicated constructor to let using borrowingLimit =0, so that we are sure we don't drop some coverage?

Reply:
+1. nil borrowingLimit means "borrow any amount". Please revert numbers.

Reply:
Sorry, my explaination was wrong, but the test was still broken. See explaination here https://github.com/kubernetes-sigs/kueue/pull/2486#discussion_r1662224680